### DISCUSSION TITLE
Plotting RHIs

### DISCUSSION THREAD

**Discussion by MohitDahliya** [ARM-DOE/pyart#1635] [Q&A] [answered]
I have radar data in PPI mode. I need to construct RHI at a time instant at a particular azimuth angle. I have constructed a pseudo RHI using radar_rhi = pyart.util.cross_section_ppi(radar,[140]) where radar is the original radar data at that time instant in PPI mode, and afterwards plotted it using pyart.graph.RadarDisplay. However, the pseudo RHI that gets plotted has an awkward shape looking like a triangle. Additionally, the features in the near vicinity of radar at upper levels are not visible, and that is where major portion of the cloud system was. Is there any alternate way of constructing RHI from PPI data, so that I don't have to face these issues?

**Comment by zssherman** [comment]
@MohitDahliya Can you share an image on what that plot looks like?

**Comment by MohitDahliya** [comment]
![image](https://github.com/user-attachments/assets/59b55d5e-03ed-46e2-8ce9-97c8c5a2e466)
I have attached the link to the figure as you aksed. Please look into it nad provide your valuable insights.

**Comment by MohitDahliya** [comment]
Hi there ! No reply for a long time. Can I perform some sort of interpolation so that I can also construct meaningful visualization in the currently empty region in the plots.

**Reply by zssherman** [reply]
@MohitDahliya Are you talking about the whitespace above the data? For an RHI your only going to get data where your elevations cover, the triangle looks like an RHIs to me or are you talking about an artifact in the data itself? If your trying to do some sort of interpolation, you could try gridding the data and do a cross section:
https://arm-doe.github.io/pyart/examples/plotting/plot_three_panel_gridmapdisplay.html#sphx-glr-examples-plotting-plot-three-panel-gridmapdisplay-py

Granted thats based on latitude and longitude.

**Reply by mgrover1** [reply, marked as answer]
The image you shared is what an RHI from a PPI should look like - there likely were not elevation scans higher up, leading a lack of data. This seems like more of a lack of data issue, not necessarily Py-ART. The empty regions are meaningful; they are where there was not data collected.

**Reply by mgrover1** [reply]
I would suggest taking a look at creating a Quasi Vertical Profile (QVP), which you can do using Py-ART https://arm-doe.github.io/pyart/API/generated/pyart.retrieve.compute_qvp.html#pyart.retrieve.compute_qvp

More on the methodology here - https://doi.org/10.1175/JTECH-D-15-0020.1

**Comment by MohitDahliya** [comment]
Given your advuce, I have decided to retain these plots and will be working with them. Thank you for taking out time and discussing.
